### PR TITLE
Fix optdata restore functionality

### LINKED ISSUE #11796: OptData sync needs to be re-enabled
The BuildTools/CLI update that migrated CoreCLR from project.json -> \*.csproj removed the support for syncing optdata packages. This blocks PGO and IBC optimizations from being applied on release builds.

### PR DESCRIPTION
**Pull request by dpodder**
Bring back optdata restore functionality following the move to 2.0 BuildTools and csproj-based CLI. Disable a harmless warning that broke the Linux build when consuming PGO counts due to -Werr.

Fix #11796 for master. We will need to get this ported to release/2.0.0 as well.

### CODE REVIEW

File: dir.props [hunk 1]
```diff
@@ -21,6 +21,13 @@
     <RoslynPackageName>Microsoft.Net.ToolsetCompilers</RoslynPackageName>
   </PropertyGroup>
 
+  <!-- Profile-based optimization data package versions -->
+  <PropertyGroup>
+    <!-- TODO: Fill in with actual versions from the right branches -->
```
Comment (dpodder) [inline dir.props]: FYI: We are generating counts packages today and I plan to resolve this TODO before merging. If that gets missed, nothing breaks, and it's a trivial follow-up PR to update this.

**Comment by dpodder** [comment]
@wtgodbe @adiaaida PTAL
/CC: @gkhanna79 @weshaggard

**Comment by gkhanna79** [comment]
Please email shiproom to bring this to 2.0.0, assuming this is merged into master.

**Comment by dpodder** [comment]
@gkhanna79 Definitely, will do.

**Comment by dpodder** [comment]
@dotnet-bot test windows_nt x86 perf

**Comment by dpodder** [comment]
@dotnet-bot test windows_nt x64 perf

**Comment by dpodder** [comment]
@dotnet-bot test Ubuntu14.04 perf

**Comment by dpodder** [comment]
@dotnet-bot test Ubuntu14.04 perf

**Comment by dpodder** [comment]
@jorive PTAL

**Comment by adiaaida** [comment]
LGTM, but do we want to have an option to skip using optdata? Like if we wanted to do a perf comparison of pgo vs non-pgo?

**Comment by dpodder** [comment]
@adiaaida thanks for the review! You're right; we do still have that functionality, `skiprestoreoptdata` when passed to build.cmd/build.sh. Doing so will skip invoking `dotnet restore` on `optdata.csproj` and as a result nothing will see any PGO or IBC counts.

**Pull request merged**